File: react-web/library/src/components/TspModal.js
import { React, h, cx, clamp, optionText, themed, starPlanetTheme } from './_shared.js';
import { TspButton } from './TspButton.js';

/**
 * TspModal – Confirmation dialog.
 *
 * Full-screen overlay with centered panel containing title, message,
 * and confirm/cancel action buttons.
 *
 * @param {Object} props
 * @param {string} [props.title] - Dialog title.
 * @param {string} [props.message] - Dialog body message.
 * @param {string} [props.confirmText='OK'] - Confirm button text.
 * @param {string} [props.cancelText='Cancel'] - Cancel button text.
 * @param {Object} [props.theme] - Theme object.
 * @param {Function} [props.onConfirm] - Confirm button handler.
 * @param {Function} [props.onCancel] - Cancel button handler.
 * @returns {React.ReactElement} A dialog overlay element.
 */

export function TspModal({ title, message, confirmText = 'OK', cancelText = 'Cancel', theme = starPlanetTheme, onConfirm, onCancel }) {
  return h(
    'div',
    { className: 'bc-modal', style: themed(theme), role: 'dialog', 'aria-modal': 'true' },
    h('section', { className: 'bc-modal__panel' },
      title && h('h3', null, title),
      message && h('p', null, message),
      h('div', { className: 'bc-modal__actions' },
        h(TspButton, { text: cancelText, variant: 'default', theme, onTap: onCancel }),
        h(TspButton, { text: confirmText, variant: 'primary', theme, onTap: onConfirm })
      )
    )
  );
}

File: ios-swiftui/library/Sources/PlanetComponents/TspModal.swift
import SwiftUI

/// Confirmation dialog — aligns with RN `TspModal` (mask + panel + dual actions).
public struct TspModal: View {
    let title: String
    let message: String
    let confirmText: String
    let cancelText: String
    let theme: StarPlanetTheme
    let onConfirm: () -> Void
    let onCancel: () -> Void

    public init(
        title: String,
        message: String,
        confirmText: String = "OK",
        cancelText: String = "Cancel",
        theme: StarPlanetTheme = .sky,
        onConfirm: @escaping () -> Void = {},
        onCancel: @escaping () -> Void = {}
    ) {
        self.title = title
        self.message = message
        self.confirmText = confirmText
        self.cancelText = cancelText
        self.theme = theme
        self.onConfirm = onConfirm
        self.onCancel = onCancel
    }

    public var body: some View {
        ZStack {
            Color(red: 0.090, green: 0.227, blue: 0.384).opacity(0.28)
                .ignoresSafeArea()
                .onTapGesture(perform: onCancel)
            VStack(alignment: .leading, spacing: 14) {
                Text(title)
                    .font(.system(size: 20, weight: .black))
                    .foregroundColor(theme.textPrimary)
                Text(message)
                    .font(.system(size: 15))
                    .foregroundColor(theme.textSecondary)
                    .fixedSize(horizontal: false, vertical: true)
                HStack(spacing: 10) {
                    TspButton(cancelText, theme: theme, action: onCancel)
                    TspButton(confirmText, variant: .primary, theme: theme, action: onConfirm)
                }
            }
            .padding(22)
            .frame(maxWidth: 420)
            .background(theme.surfaceRaised)
            .overlay(RoundedRectangle(cornerRadius: 24).stroke(theme.borderDefault, lineWidth: 1))
            .clipShape(RoundedRectangle(cornerRadius: 24))
            .padding(.horizontal, 18)
        }
    }
}
